framework/src/problems/SubProblem.C

```diff
@@ -886,6 +886,29 @@ SubProblem::reinitLowerDElemRef(const Elem * elem,
   systemBaseAuxiliary().reinitLowerD(tid);
 }
 
+void
+SubProblem::reinitLowerDElemDualRef(const Elem * elem,
+                                    const std::vector<Point> * const pts,
+                                    const std::vector<Real> * const weights,
+                                    THREAD_ID tid)
+{
+  // - Set our _current_lower_d_elem for proper dof index getting in the moose variables
+  // - Reinitialize all of our lower-d FE objects so we have current phi, dphi, etc. data
+  assembly(tid).reinitLowerDElemDualRef(elem, pts, weights);
+
+  // Actually get the dof indices in the moose variables
+  systemBaseNonlinear().prepareLowerD(tid);
+  systemBaseAuxiliary().prepareLowerD(tid);
+
+  // With the dof indices set in the moose variables, now let's properly size
+  // our local residuals/Jacobians
+  assembly(tid).prepareLowerD();
+
+  // Let's finally compute our variable values!
+  systemBaseNonlinear().reinitLowerD(tid);
+  systemBaseAuxiliary().reinitLowerD(tid);
+}
+
```

Comment:
Yea there's a lot of code duplication here. Again, I think we can design this such that a call to `Assembly::reinitLowerDElemRef` initializes both standard and dual data structures